packages/middleware/src/apiClientFactory/applyContextToApi.ts

```diff
@@ -36,6 +38,17 @@ const applyContextToApi = <
         const extendQuery = createExtendQuery(context);
         const transformedArgs = await hooks.before({ callName, args });
         const apiClientContext = { ...context, extendQuery };
+        if (isCustomEndpointHandler(fn)) {
+          if (context.res.locals?.alokai?.metadata?.scope) {
+            context.res.locals.alokai.metadata.scope.extensionName =
+              fn._extensionName;
+          } else {
+            const logger = getLogger(context.res);
+            logger.warning(
+              "Local Alokai's metadata object is missing. Check your source code or contact Alokai's team."
```

Comment:
maybe
```
Alokai's metadata object is missing in 'res.locals.alokai'. This could impact extension-specific functionality, 
  such as scope tracking. Ensure 'res.locals.alokai.metadata' is properly initialized. 
  Verify middleware setup or contact the Alokai team for assistance.
  Steps to troubleshoot:
  Check if .... bla bla....
```

current message is very unclear